### ISSUE TITLE
Why are st_mode restored only for memfd, socket, and reg-file, but not for pipe, evenetfd...?

### ISSUE THREAD

**Issue by cuiyunhui-1** [checkpoint-restore/criu#2281]
**Description**

**Steps to reproduce the issue:**
1.
2.
3.

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

**CRIU logs and information:**

<details><summary>CRIU full dump/restore logs:</summary>
<p>

```
(paste your output here)
```

</p>
</details>

<details><summary>Output of `criu --version`:</summary>
<p>

```
(paste your output here)
```

</p>
</details>

<details><summary>Output of `criu check --all`:</summary>
<p>

```
(paste your output here)
```

</p>
</details>

**Additional environment details:**

**Comment by cuiyunhui-1** [comment]
@emmir

**Comment by cuiyunhui-1** [comment]
@osctobe

**Comment by avagin** [comment]
I am not sure that I understand the question. What is st_mode in this context? Where do we restore it for memfd, socket, reg-files?

**Comment by cuiyunhui-1** [comment]
> I am not sure that I understand the question. What is st_mode in this context? Where do we restore it for memfd, socket, reg-files?

static int memfd_open_inode_nocache(struct memfd_restore_inode *inode)
{
...

	fd = memfd_create(mie->name, flags);

	if (restore_memfd_shmem_content(fd, mie->shmid, mie->size))
		goto out;

	if (mie->has_mode)
		ret = cr_fchperm(fd, mie->uid, mie->gid, mie->mode);
	else
		ret = cr_fchown(fd, mie->uid, mie->gid);
...

}

**Comment by cuiyunhui-1** [comment]
@avagin

**Comment by avagin** [comment]
> What I want to know is why CRIU doesn’t uniformly perform st_mode on all fd types.

You are right, we may need to restore mode, uid, gid for mentioned. However, we have to decide whether we need to do that for each file type separately. For example, there is no need to do this for eventfd since, in the kernel, all eventfd instances are linked to the same anonymous inode.:
https://elixir.bootlin.com/linux/latest/source/fs/eventfd.c#L418
https://elixir.bootlin.com/linux/latest/source/fs/anon_inodes.c#L147
https://elixir.bootlin.com/linux/latest/source/fs/anon_inodes.c#L97

Secondly, for certain file types, it is less critical since users typically do not reopen them through file systems.

**Comment by cuiyunhui-1** [comment]
@avagin okay，thank you！
